engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -235,6 +237,53 @@ Stateful doConvertText(
     }
   }
 
+  @Specialization(
+      guards = {
+        "!methods.hasFunctionalDispatch(_this)",
+        "!methods.hasSpecialDispatch(_this)",
+        "getPolyglotCallType(_this, symbol.getName(), interop) == CONVERT_TO_DATE"
+      })
+  Stateful doConvertDate(
+      VirtualFrame frame,
+      Object state,
+      UnresolvedSymbol symbol,
+      Object _this,
+      Object[] arguments,
+      @CachedLibrary(limit = "10") MethodDispatchLibrary methods,
+      @CachedLibrary(limit = "1") MethodDispatchLibrary dateDispatch,
+      @CachedLibrary(limit = "10") InteropLibrary interop,
+      @Cached("wrapDateInAtom(interop)") java.util.function.Function<Object,Atom> wrapDate
```

Comment:
I think this would be more elegant as a separate node, not a bare function

Reply:
OK, done in: 94221c200